rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -97,15 +97,21 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None):
         if topic_type is not None and topic_type != msg_type_string:
             raise TypeConflictException(topic, topic_type, msg_type_string)
 
+        # Get publishers info
+        publishers_info = node_handle.get_publishers_info_by_topic(topic)
+
+        # Select QoS
+        default_qos_profile = 10
+        qos_profile = publishers_info[0].qos_profile if publishers_info else default_qos_profile
```

Comment:
念の為確認です
基本的には同じトピックに複数publisherはいないという想定でpublishers_info[0]になっているということですよね？

Reply:
Yes, if we'd like to support more complex cases, I think it can be extended like ros2cli.
https://github.com/ros2/ros2cli/blob/d974fdefb0e8f0af095ac6e211e5b17b0115974a/ros2topic/ros2topic/verb/echo.py#L140-L177

Reply:
I think the only case like the above that is defined as an interface is diag, but since it doesn't bridge, I think it's okay.